### PR TITLE
ICU-21635 Unicode 14 data files 20210820, line break LB30b.2

### PR DESCRIPTION
**Pull request by markusicu**
- Unicode 14 data files as of 2021-aug-19 with a LineBreakTest.txt fix from aug20
- additional line break rule: second part of LB30b, do not break between a *potential emoji* and an emoji modifier

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21635
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/rbbi/RBBITestMonkey.java [hunk 1]
```diff
@@ -681,51 +681,52 @@ int   next(int prevPos) {
...
-        UnicodeSet  fJV;
-        UnicodeSet  fJT;
-        UnicodeSet  fRI;
-        UnicodeSet  fXX;
-        UnicodeSet  fEB;
-        UnicodeSet  fEM;
-        UnicodeSet  fZWJ;
-        UnicodeSet  fOP30;
-        UnicodeSet  fCP30;
+        XUnicodeSet fBK;
```
Comment (FrankYFTang) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/test/rbbi/RBBITestMonkey.java:684]: Why is this UnicodeSet to XUnicodeSet needed? Could you add a comment on the top?

**Review by FrankYFTang — APPROVED** [review]
Thanks for working on this. Please add that comment about XUnicodeSet

File: icu4j/main/tests/core/src/com/ibm/icu/dev/test/rbbi/RBBITestMonkey.java (hunk 1 above)
Comment (markusicu) [reply icu4j/main/tests/core/src/com/ibm/icu/dev/test/rbbi/RBBITestMonkey.java:705]: Done: Pulled up the declaration of class XUnicodeSet, and added another comment for why I changed the variable declaration type. Use `--amend` to keep it to one commit. Please re-approve.

**Review by FrankYFTang — DISMISSED** [review]

**Review by aheninger — APPROVED** [review]
Looked mostly at the break iterator related changes.

File: icu4c/source/test/intltest/rbbimonkeytest.cpp [hunk 2]
```diff
@@ -661,7 +661,7 @@ void RBBIMonkeyImpl::join() {
 
 
 #define MONKEY_ERROR(msg, index) UPRV_BLOCK_MACRO_BEGIN { \
-    IntlTest::gTest->errln("%s:%d %s at index %d. Parameters to reproduce: @rules=%s,seed=%u,loop=1,verbose ", \
+    IntlTest::gTest->errln("%s:%d %s at index %d. Parameters to reproduce: @\"rules=%s,seed=%u,loop=1,verbose\"", \
```
[the remark this answers is not in the document]
Comment (markusicu) [reply icu4c/source/test/intltest/rbbimonkeytest.cpp]: Ah, good to know. I grep’ed and saw both versions, so added the quotes to be safe. Reverted/amended once more to keep this clean. PTAL

**Review by aheninger — APPROVED** [review]

**Pull request merged**